res/controllers/Hercules-DJ-Console-4-Mx-scripts.js

```diff
@@ -0,0 +1,622 @@
+        //Tell shift not to change state.
+        Hercules4Mx.shiftUsed = true;
+    }
+    switch (control) {
+        // 1 to 6 and 0x21 to 0x26 are the 6 left deck and right deck buttons respectively, non shifted.
+        // Loop buttons
+        case 0x01:
+        case 0x21: // K1, beatloop 1/ Loop in
+            if (value) {
+                //Loop in in "shift-pressed" mode, beatloop of 1 beat otherwise
+                if (Hercules4Mx.shiftPressed) {
+                    engine.setValue(group, "beatloop_0.125_toggle", 1);
+                } else {
+                    engine.setValue(group, "beatloop_0.5_toggle", 1);
+                }
+            }
+            break;
+        case 0x02:
+        case 0x22: // K2, beatloop 2/ Loop out
+            if (value) {
+                //Loop out in "shift-pressed" mode, beatloop of 2 beat otherwise
+                if (Hercules4Mx.shiftPressed) {
+                    engine.setValue(group, "beatloop_0.25_toggle", 1);
+                } else {
+                    engine.setValue(group, "beatloop_1_toggle", 1);
+                }
+            }
+            break;
+        case 0x03:
+        case 0x23: // K3, beatloop 4/ Reloop/Exit
+            if (value) {
+                //enable/disable loop in "shift-pressed" mode, beatloop of 2 beat otherwise
+                if (Hercules4Mx.shiftPressed) {
+                    engine.setValue(group, "reloop_exit", 1);
+                } else {
+                    engine.setValue(group, "beatloop_2_toggle", 1);
+                }
+            }
+            break;
+        case 0x04:
+        case 0x24: // K4, beatloop 8/ Reloop/Exit
+            if (value) {
+                //enable/disable loop in "shift-pressed" mode, beatloop of 4 beat otherwise
+                if (Hercules4Mx.shiftPressed) {
+                    engine.setValue(group, "reloop_exit", 1);
+                } else {
+                    engine.setValue(group, "beatloop_4_toggle", 1);
+                }
+            }
+            break;
+            // Reverse play buttons
+        case 0x05:
+        case 0x25: // K5, reverse play
+            engine.setValue(group, "reverse", (value) ? 1 : 0);
+            engine.setValue(group, "reverse", (value) ? 1 : 0);
+            break;
+        case 0x06:
+        case 0x26: // K6, reverse roll play
+            engine.setValue(group, "reverseroll", (value) ? 1 : 0);
+            engine.setValue(group, "reverse", (value) ? 1 : 0);
+            break;
+
+            // 7 to 0xC and 0x27 to 0x3C are the 6 left deck and right deck buttons respectively, with shift enabled.
+            // Hotcue buttons:
+        case 0x07:
+        case 0x27: // K7 Hotcue 1 activate/ clear
+            if (Hercules4Mx.shiftPressed) {
+                if (value) {
+                    engine.setValue(group, "hotcue_1_clear", 1);
+                }
+            } else engine.setValue(group, "hotcue_1_activate", (value) ? 1 : 0);
```

Comment:
The Controller manual describes the 12 possible Effect button actions as:
1 - Loop begin
2 - Loop end, loop exit.
3 to 6 - Hotcues 1 to 4
7 - Record
8 - Play recorded  (These two are like a dynamic sampler, that replays a part of the loaded track)
9 to 12 - Four effects.

What is actually mapped in virtual DJ is:
1 - Loop begin
2 - Loop end, loop exit.
3 - Flanger
4 - Hotcue 2
5 - Beatgrid (it is some kind of effect that retriggers audio following a user-configurable beat grid). Like Tatata-tata.
6 - Bitcrunch or something like that..
7 - Record
8 - Play recorded  (These two are like a dynamic sampler, that replays a part of the loaded track)
9 - Hotcue 1
10 - Another weird effect
11 and 12 Hotcues 3 and 4.

What I've mapped to Mixxx:
1 to 4 beatloops. Loop out/reloop can be triggered with shift+button 3 or 4.
5 and 6 reverse play
7 to 10 hotcues 1 to 4. Each Hotcue can be cleared with shift+button
11 and 12 effect Unit 1 and 2.

Shift button reworded in the wiki.
-------------------------

Tray-icon referes to the application that configures the controller, installed with the driver on Windows and Mac, yes.
Anyway, I've removed the ability to change the midi channel. 
It had no sense to offer it as a parameter because it required changing the mapping too.

Ordering and variable type modifications as suggested.

Indentation: I preffer to add the && before the line instead of the end, but that format checker page told me to put it at the end. I've added more indentation now.
----------------------
The Files/Folder and sidebar/list navigation is completely controlled in the script. Mixxx simply offers ways to navigate and i decide from 
the script what I am navigating and the user controls it with the files and folder buttons.
The only small glitch i found is that if i change the element on the sidebar and then go to files, the first up/down movement doesn't really move because the cursor
is not placed anywhere and that first button press puts it at the beginning.